elpis/engines/common/objects/transcription.py

```diff
@@ -54,11 +67,17 @@ def state(self):
     @property
     def has_been_transcribed(self):
         return self.config['has_been_transcribed']
-    
+
     @property
     def exporter(self):
         return self._exporter
 
+    def build_stage_status(self, stage_names: Dict[str, str]):
+        for stage_file, stage_name in stage_names.items():
+            stage_status = self.config['stage_status']
+            stage_status.update({stage_file: {'name': stage_name, 'status': 'ready', 'message': ''}})
+            self.config['stage_status'] = stage_status
```

Comment:
Note on efficiency. Not sure how many stages there are but if there are over 10 this will make the harddrive work a little hard. The `self.config['_______']` reads from disk and writes to disk every time it is used. May want to just have the `update` line in the for-loop and config read/writes outside so they only happen once.

```suggestion
        stage_status = self.config['stage_status']
        for stage_file, stage_name in stage_names.items():
            stage_status.update({stage_file: {'name': stage_name, 'status': 'ready', 'message': ''}})
        self.config['stage_status'] = stage_status
```

Reply:
good point, given that we currently only have the few Kaldi steps, let's leave it as is. If the ESPnet work ends up with more fine-grained stages let's update then.